src/lcm/dispatchers.py

```diff
@@ -191,21 +191,81 @@ def allow_kwargs(func):
 
     @functools.wraps(func)
     def allow_kwargs_wrapper(*args, **kwargs):
-        parameters = list(inspect.signature(func).parameters)
-
-        positional = list(args) if args is not None else []
-
-        kwargs = {} if args is None else kwargs
-        if len(args) + len(kwargs) != len(parameters):
+        parameters = inspect.signature(func).parameters
+
+        # Retrieve keyword-only arguments and remove from kwargs
+        keyword_only = {
+            k: v
+            for k, v in kwargs.items()
+            if k in parameters and parameters[k].kind == inspect.Parameter.KEYWORD_ONLY
+        }
```

Comment:
It wasn't done before but I think we should move everything we can out of the wrapper and into the decorator. I.e. reading the signature and creating a list of keyword only arguments. Just to make sure we don't get a runtime cost.